src/main/java/oof/Ui.java

```diff
@@ -0,0 +1,757 @@
+                " 8", " 9", "10", "11", "12", "13", "14", "15",
+                "16", "17", "18", "19", "20", "21", "22", "23",
+                "24", "25", "26", "27", "28", "29", "30", "31"};
+        ArrayList<String> calendarDates = new ArrayList<>();
+        for (int i = 1; i <= yearMonth.lengthOfMonth(); i++) {
+            calendarDates.add(date[i]);
+        }
+
+        DayOfWeek firstDayOfMonth = yearMonth.atDay(DAY_FIRST).getDayOfWeek();
+        if (firstDayOfMonth != DayOfWeek.SUNDAY) {
+            for (int dayCount = 0; dayCount < firstDayOfMonth.getValue(); dayCount++) {
+                calendarDates.add(INDEX_START_OF_ARRAY, date[INDEX_SPACE]);
+            }
+        }
+
+        while (calendarDates.size() % DAYS_IN_WEEK != 0) {
+            calendarDates.add(date[INDEX_SPACE]);
+        }
+
+        int numberOfWeeks = (int) Math.ceil((double) calendarDates.size() / DAYS_IN_WEEK);
+        for (int week = 0; week < numberOfWeeks; week++) {
+            int dayIndex = week * DAYS_IN_WEEK;
+            System.out.print("|");
+            int calendarRows = WEEKS_IN_MONTH;
+            for (int day = 0; day < DAYS_IN_WEEK; day++) {
+                if (!calendarDates.get(dayIndex + day).equals("  ")) {
+                    int currentDay = Integer.parseInt(calendarDates.get(dayIndex + day).trim());
+                    if (calendar.get(currentDay).size() > calendarRows) {
+                        calendarRows = calendar.get(currentDay).size();
+                    }
+                }
+                System.out.print(" " + calendarDates.get(dayIndex + day) + "                  |");
+            }
+            System.out.println();
+            printCalendarDetails(calendar, calendarDates, dayIndex, calendarRows);
+            System.out.println("-------------------------------------------------------------------------------------"
+                    + "----------------------------------------------------------------------");
+        }
+    }
+
+    /**
+     * Prints calendar details.
+     *
+     * @param calendar      ArrayList containing task information for current month.
+     * @param calendarDates ArrayList containing dates for current month.
+     * @param dayIndex      Offset for current day.
+     * @param calendarRows  Number of rows to be printed for current week.
+     */
+    public void printCalendarDetails(ArrayList<ArrayList<String[]>> calendar, ArrayList<String> calendarDates,
+                                     int dayIndex, int calendarRows) {
+        for (int row = 0; row < calendarRows; row++) {
+            System.out.print("|");
+            for (int day = 0; day < DAYS_IN_WEEK; day++) {
+                String dayString = calendarDates.get(dayIndex + day).trim();
+                if (dayString.equals("") || calendar.get(Integer.parseInt(dayString)).size() <= row) {
+                    System.out.print("                     |");
```

Comment:
You can find a way to print `x` number of spaces instead of hard-coding it.

This creates problems when you are running test cases.